packages/speech/src/index.js

```
@@ -0,0 +1,804 @@
+    ],
+    packageJson: require('../package.json')
+  };
+
+  common.GrpcService.call(this, config, options);
+}
+
+util.inherits(Speech, common.GrpcService);
+
+/**
+ * The endpointer types that the Speech API will return while processing a
+ * {module:speech#createRecognizeStream} request. You can track the progress of
+ * audio recognition by comparing the `data.endpointerType` property with these
+ * values.
+ *
+ *   - `Speech.endpointerTypes.ENDPOINTER_EVENT_UNSPECIFIED`: No endpointer
+ *     event specified.
+ *   - `Speech.endpointerTypes.START_OF_SPEECH`: Speech has been detected in the
+ *     audio stream.
+ *   - `Speech.endpointerTypes.END_OF_SPEECH`: Speech has ceased to be detected
+ *     in the audio stream.
+ *   - `Speech.endpointerTypes.END_OF_AUDIO`: The end of the audio stream has
+ *     been reached and it is being processed.
+ *   - `Speech.endpointerTypes.END_OF_UTTERANCE`: This event is only sent when
+ *     `config.singleUtterance` passed to {module:speech#createRecognizeStream}
+ *     is `true`. It indicates that the server has detected the end of the
+ *     user's speech utterance and expects no additional speech. Therefore, the
+ *     server will not process additional audio. The client should stop sending
+ *     additional audio data.
+ *
+ * @type {object}
+ */
+Speech.endpointerTypes =
+Speech.prototype.endpointerTypes = {
+  END_OF_AUDIO: 'END_OF_AUDIO',
+  END_OF_SPEECH: 'END_OF_SPEECH',
+  END_OF_UTTERANCE: 'END_OF_UTTERANCE',
+  ENDPOINTER_EVENT_UNSPECIFIED: 'ENDPOINTER_EVENT_UNSPECIFIED',
+  START_OF_SPEECH: 'START_OF_SPEECH'
+};
+
+/**
+ * Guess the audio encoding from the file's extension.
+ *
+ * @resource [AudioEncoding API Documentation]{@link https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#audioencoding}
+ * @private
+ *
+ * @param {string} filename - The name of the file.
+ * @returns {string} The audio encoding.
+ */
+Speech.detectEncoding_ = function(filename) {
```

Should we throw an error here for unknown encoding types..? Or is that not really a concern?